csv-generator/csv_generator/consumers/consumer.py

```diff
@@ -33,22 +33,43 @@ def output_file(self) -> str:
             self._output_file = os.path.join(self.output_dir, '{0}_{1}'.format(self.create_date, self.base_file_name))
         return self._output_file
 
-    def process(self, element: 'lxml.etree.ElementTree', xml_file_name: str) -> None:
+    def process(self, element: 'lxml.etree.ElementTree', source_file_name: str) -> None:
         """Parse target `lxml.etree.ElementTree` object, extract required data and
         write data row to output_file.
 
         :param element: class: `lxml.etree.ElementTree`
-        :param xml_file_name:
+        :param source_file_name:
         :return:
         """
-        pass
+        raise NotImplementedError()
+
+    # https://github.com/elifesciences/data-pipeline/issues/86
+    def source_file_path(self, row):
```

Comment:
This is more or less what I thought a similar method would look like:
```python
def source_file_path(zip_file_name, file_name):
    return zip_file_name + "/" + source_file_name if zip_file_name else file_name
```

I can see the description regarding the CSV could be confusing. It's just that source CSVs are never part of a zip file. If they were, we would refer to the zip file in the same way.

Reply:
I'm not precious about it, if I've misinterpreted your logic please correct the code. 

In fact, for most of these trivial tickets, I would suggest just burning through them yourself. If you're struggling with anything specifically I'm happy to help as I would help anybody, but with Sean leaving you're now the de-facto owner of this code.

Reply:
Apologies, should have made the ticket more clear. I thought it was a good ticket to get familiar with the a cross section of the code base.

I refactored / simplified the _csv-generator_ changes slightly.